File: cpp/hewan.cpp
#pragma once
#include <bits/stdc++.h>
#include "makanan.cpp"
using namespace std;

class hewan 
{
private:
    string famili;
    string habitat;
    string jenisMakanan;
    string caraBerkembangbiak;
    int harapanHidup;
    makanan food;
public:
    hewan() {}
    hewan(string famili, string habitat, string jenisMakanan, string caraBerkembangbiak, int harapanHidup, makanan food) 
    {
        this->famili = famili;
        this->habitat = habitat;
        this->jenisMakanan = jenisMakanan;
        this->caraBerkembangbiak = caraBerkembangbiak;
        this->harapanHidup = harapanHidup;
        this->food = food;
    }

    // Setter
    void setFamili(string f) { famili = f; }
    void setHabitat(string h) { habitat = h; }
    void setJenisMakanan(string jm) { jenisMakanan = jm; }
    void setCaraBerkembangbiak(string cb) { caraBerkembangbiak = cb; }
    void setHarapanHidup(int hh) { harapanHidup = hh; }
    void setMakanan(makanan m) { food = m; }

    // Getter
    string getFamili() { return famili; }
    string getHabitat() { return habitat; }
    string getJenisMakanan() { return jenisMakanan; }
    string getCaraBerkembangbiak() { return caraBerkembangbiak; }
    int getHarapanHidup() { return harapanHidup; }
    makanan getMakanan() { return food; }

    virtual void print() 
    {
        cout << "Famili             : " << getFamili() << endl;
        cout << "Habitat            : " << getHabitat() << endl;
        cout << "Jenis Makanan      : " << getJenisMakanan() << endl;
        cout << "Cara Berkembangbiak: " << getCaraBerkembangbiak() << endl;
        cout << "Harapan Hidup      : " << getHarapanHidup() << " tahun" << endl;
        getMakanan().print();
    }
};

File: py/hewan.py
from makanan import makanan

class hewan:
    def __init__(self, famili="", habitat="", jenis_makanan="", cara_berkembangbiak="", harapan_hidup=0, food=None):
        self.famili = famili
        self.habitat = habitat
        self.jenis_makanan = jenis_makanan
        self.cara_berkembangbiak = cara_berkembangbiak
        self.harapan_hidup = harapan_hidup
        self.food = food if food else makanan()

    # Setter
    def set_famili(self, f):
        self.famili = f

    def set_habitat(self, h):
        self.habitat = h

    def set_jenis_makanan(self, jm):
        self.jenis_makanan = jm

    def set_cara_berkembangbiak(self, cb):
        self.cara_berkembangbiak = cb

    def set_harapan_hidup(self, hh):
        self.harapan_hidup = hh

    def set_makanan(self, m):
        self.food = m

    # Getter
    def get_famili(self):
        return self.famili

    def get_habitat(self):
        return self.habitat

    def get_jenis_makanan(self):
        return self.jenis_makanan

    def get_cara_berkembangbiak(self):
        return self.cara_berkembangbiak

    def get_harapan_hidup(self):
        return self.harapan_hidup

    def get_makanan(self):
        return self.food

    def print(self):
        print(f"Famili             : {self.get_famili()}")
        print(f"Habitat            : {self.get_habitat()}")
        print(f"Jenis Makanan      : {self.get_jenis_makanan()}")
        print(f"Cara Berkembangbiak: {self.get_cara_berkembangbiak()}")
        print(f"Harapan Hidup      : {self.get_harapan_hidup()} tahun")
        self.get_makanan().print()